### PR TITLE
ts: basic documentation for utility scripts

### PR DESCRIPTION
**Pull request by mshabunin**
* Added documentation strings for all scripts in `modules/ts/misc`
* Changed executable bit for several scripts (enabled for some, disabled for others)

### CODE REVIEW

File: modules/ts/misc/perf_tests_timing.py [hunk 1]
```diff
@@ -1,4 +1,22 @@
 #!/usr/bin/env python
+""" Prints total execution time and number of total/failed tests.
+
+This script uses XML test log to produce basic runtime statistics in a text or HTML table.
```
Comment (asmorkalov) [inline modules/ts/misc/perf_tests_timing.py:4]: I propose to add some words on `--gtest_ouput` usage to highlight the source of xml and statistics.

File: modules/ts/misc/report.py [hunk 2]
```diff
@@ -1,4 +1,32 @@
 #!/usr/bin/env python
+""" Print performance test run statistics.
+
+This script produces configurable performance report tables in text and HTML formats. It allows to filter test cases by name and parameter string and select specific performance metrics columns. One or multiple test results can be used for input.
```
Comment (asmorkalov) [inline modules/ts/misc/report.py:4]: The same on `--gtest_output`.

File: modules/ts/misc/summary.py [hunk 3]
```diff
@@ -1,4 +1,31 @@
 #!/usr/bin/env python
+""" Format performance test results and compare metrics between test runs
+
+This script allows to compare performance data collected during separate test runs and present it in a text, Markdown or HTML table.
```
Comment (asmorkalov) [inline modules/ts/misc/summary.py:4]: Need some reference to OpenCV perf tests and `--gtest_output` option.

File: modules/ts/misc/trace_profiler.py [hunk 4]
```diff
@@ -1,3 +1,26 @@
+#!/usr/bin/env python
+""" Parse OpenCV trace logs and present summarized statistics in a table
+
+### Options
+
+./trace_profiler.py <TraceLogFile> <num>
+
+<TraceLogFile>  - usually OpenCVTrace.txt
```
Comment (asmorkalov) [inline modules/ts/misc/trace_profiler.py:8]: Some instructions to get OpenCVTrace.txt are required or link to wiki, if it's not trivial.

File: modules/ts/misc/perf_tests_timing.py (hunk 1 above)
Comment (mshabunin) [reply modules/ts/misc/perf_tests_timing.py:4]: Done.

File: modules/ts/misc/report.py (hunk 2 above)
Comment (mshabunin) [reply modules/ts/misc/report.py:4]: Done.

File: modules/ts/misc/trace_profiler.py (hunk 4 above)
Comment (mshabunin) [reply modules/ts/misc/trace_profiler.py:13]: Added.

**Review by asmorkalov — APPROVED** [review]

File: modules/ts/misc/perf_tests_timing.py [hunk 5]
```diff
@@ -78,7 +100,7 @@
             suit_time = 0
             suit_num = 0
             fails_num = 0
-            for name in sorted(test_cases.iterkeys(), key=alphanum_keyselector):
+            for name in sorted(test_cases, key=alphanum_keyselector):
```
Comment (asmorkalov) [inline modules/ts/misc/perf_tests_timing.py:103]: With Python 2 it does not crash, but produce broken report. For 3.4 it's important to presume 2.7 support.
```
python2 ../opencv-3.4/modules/ts/misc/perf_tests_timing.py ./imgproc_perf.xml          

Overall time: 0.79 min

Module  Testsuit Time (min) Num of tests Failed
        r                                      
        e                                      
        s                                      
        i                                      
        z                                      
        e                                      
        D                                      
        o                                      
        w                                      
        n                                      
        L                                      
        i                                      
        n                                      
        e                                      
        a                                      
        r                                      
        N                                      
        o                                      
        n                                      
        E                                      
imgproc x          0.304         60            
        a                                      
        c                                      
        t                                      
        :                                      
        :                                      
        M                                      
        a                                      
        t                                      
        I                                      
        n                                      
        f                                      
        o                                      
        _                                      
        S                                      
        i                                      
        z                                      
        e                                      
        P                                      
        a                                      
        i                                      
        r                                      
        r                                      
        e                                      
        s                                      
        i                                      
        z                                      
        e                                      
        U                                      
        p                                      
        L                                      
        i                                      
        n                                      
        e                                      
        a                                      
        r                                      
        N                                      
        o                                      
        n                                      
        E                                      
        x                                      
imgproc a          0.285         24            
        c                                      
        t                                      
        :                                      
        :                                      
        M                                      
        a                                      
        t                                      
        I                                      
        n                                      
        f                                      
        o                                      
        _                                      
        S                                      
        i                                      
        z                                      
        e                                      
        P                                      
        a                                      
        i                                      
        r                                      
        r                                      
        e                                      
        s                                      
        i                                      
        z                                      
        e                                      
        D                                      
        o                                      
        w                                      
        n                                      
        L                                      
        i                                      
        n                                      
        e                                      
        a                                      
        r                                      
        :                                      
imgproc :          0.108         20            
        M                                      
        a                                      
        t                                      
        I                                      
        n                                      
        f                                      
        o                                      
        _                                      
        S                                      
        i                                      
        z                                      
        e                                      
        _                                      
        S                                      
        i                                      
        z                                      
        e                                      
        r                                      
        e                                      
        s                                      
        i                                      
        z                                      
        e                                      
        U                                      
        p                                      
        L                                      
        i                                      
        n                                      
        e                                      
        a                                      
        r                                      
        :                                      
        :                                      
imgproc M          0.096         8             
        a                                      
        t                                      
        I                                      
        n                                      
        f                                      
        o                                      
        _                                      
        S                                      
        i                                      
        z                                      
        e                                      
        _                                      
        S                                      
        i                                      
        z                                      
        e
```

Comment (mshabunin) [reply modules/ts/misc/perf_tests_timing.py]: Apparently the problem was not with `.iterkeys()`, it is the script `perf_tests_timing.py`. It does not work in _python2_ with current _3.4_ branch as well.

**Pull request merged**